.clang-tidy

```diff
@@ -90,6 +93,7 @@ Checks: [
   '-misc-non-private-member-variables-in-classes',
   '-misc-confusable-identifiers', # useful but slooo
   '-misc-use-anonymous-namespace',
+  '-misc-use-internal-linkage',
```

Comment:
Based on the [docs](https://clang.llvm.org/extra/clang-tidy/checks/misc/use-internal-linkage.html), this warning seems useful. If there is a reasonable number of occurrences (e.g. < 20), it IMHO makes sense to have the warning enabled.

Reply:
At the moment it looks like a ton of errors for all the rules I've disabled; I'll recheck, I want to disable as little `tidy` rules as possible, but at the same time I want to keep this PR small and readable. I may do another `clang-tidy -fix` pass, but it already broke our code (in that huge PR) to the point of subtle errors caught only by ASAN.